compiler/rustc_trait_selection/src/traits/select/mod.rs

```diff
@@ -199,16 +201,36 @@ enum BuiltinImplConditions<'tcx> {
     Ambiguous,
 }
 
+#[derive(Copy, Clone)]
+pub enum TreatInductiveCycleAs {
```

Comment:
nit: add comment that one variant is the correct one while the other is used for back-compat reasons.